### PR TITLE
Feature/terminal job

### PR DESCRIPTION
**Pull request by vipin-yadav-webkul**
**Add New Feature**

-  Now You can execute the export and import job using jobId with the user email Id

```
php artisan unopim:queue:work {JobId} {userEmailId}
```

**Example**

```
php artisan unopim:queue:work 1 admin@example.com
```

### CODE REVIEW

File: packages/Webkul/DataTransfer/src/Console/JobExecuteCommand.php [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+
+            return 1;
+        }
+
+        try {
+            $queueName = $this->dispatchJob($jobInstance, $userId);
+
+            $this->listenForEvents();
+
+            $connectionName = $this->argument('connection')
```
Comment (devansh-pal-webkul) [inline packages/Webkul/DataTransfer/src/Console/JobExecuteCommand.php:101]: Either add in a single line or indent once this line 
```php
            $connectionName = $this->argument('connection')
                ?: $this->laravel['config']['queue.default'];
```

or 
```php
            $connectionName = $this->argument('connection') ?: $this->laravel['config']['queue.default'];
```

File: packages/Webkul/DataTransfer/src/Console/JobExecuteCommand.php [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+    }
+
+    /**
+     * Get the current date / time.
+     */
+    protected function now(): Carbon
+    {
+        $queueTimezone = $this->laravel['config']->get('queue.output_timezone');
+
+        if ($queueTimezone &&
```
Comment (devansh-pal-webkul) [inline packages/Webkul/DataTransfer/src/Console/JobExecuteCommand.php]: Fix alignment for this if condition either in multi line or in a single line.

File: packages/Webkul/DataTransfer/src/Console/JobExecuteCommand.php [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+            $this->logFailedJob($event);
+        });
+    }
+
+    /**
+     * Write the status output for the queue worker.
+     *
+     * @return void
+     */
+    protected function writeOutput(Job $job, string $status)
```
Comment (devansh-pal-webkul) [inline packages/Webkul/DataTransfer/src/Console/JobExecuteCommand.php]: Add return type.

File: packages/Webkul/DataTransfer/src/Queue/Worker.php [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        if ($supportsAsyncSignals = $this->supportsAsyncSignals()) {
+            $this->listenForSignals();
+        }
+
+        while (true) {
+            if (isset($this->resetScope)) {
+                ($this->resetScope)();
+            }
+
+            // First, we will attempt to get the next job off of the queue. We will also
```
Comment (devansh-pal-webkul) [inline packages/Webkul/DataTransfer/src/Queue/Worker.php:24]: Use multi line comments.

**Review by devansh-pal-webkul — CHANGES_REQUESTED** [review]
@vipin-yadav-webkul , please make the requested changes.

File: packages/Webkul/DataTransfer/src/Queue/Worker.php (hunk 4 above)
Comment (vipin-yadav-webkul) [reply packages/Webkul/DataTransfer/src/Queue/Worker.php:24]: I have to follow Laravel coding standard

```
https://github.com/laravel/framework/blob/11.x/src/Illuminate/Queue/Worker.php#L143
```

File: packages/Webkul/DataTransfer/src/Console/JobExecuteCommand.php (hunk 1 above)
Comment (vipin-yadav-webkul) [reply packages/Webkul/DataTransfer/src/Console/JobExecuteCommand.php:101]: resolved

File: packages/Webkul/DataTransfer/src/Console/JobExecuteCommand.php (hunk 2 above)
Comment (vipin-yadav-webkul) [reply packages/Webkul/DataTransfer/src/Console/JobExecuteCommand.php]: resolved

File: packages/Webkul/DataTransfer/src/Console/JobExecuteCommand.php [hunk 5]
```diff
@@ -0,0 +1,304 @@
...
+
+    /**
+     * Get the current date / time.
+     */
+    protected function now(): Carbon
+    {
+        $queueTimezone = $this->laravel['config']->get('queue.output_timezone');
+
+        if ($queueTimezone
+            && $queueTimezone !== $this->laravel['config']->get('app.timezone')
```
Comment (devansh-pal-webkul) [inline packages/Webkul/DataTransfer/src/Console/JobExecuteCommand.php:298]: Not resolved.This should be like this 
```php
        if (
            $queueTimezone
            && $queueTimezone !== $this->laravel['config']->get('app.timezone')
         ) {
```

File: packages/Webkul/DataTransfer/src/Queue/Worker.php (hunk 4 above)
Comment (devansh-pal-webkul) [reply packages/Webkul/DataTransfer/src/Queue/Worker.php:24]: Ok, since laravel framework files have comments the same way there is no need for this.

**Review by devansh-pal-webkul — APPROVED** [review]

**Pull request merged**